feat(vtkImageData.computeHistogram): added number of pixels involved

jadh4v opened the pull request:
Added number of pixels involved in the computations to the return array of `computeHistogram()`.

Added the number of pixels involved in the calculation of the various values as part of the return
object, to fulfill an application side requirement. It also makes sense to have this information returned
since the function takes a mask/stencil style function as input.

Corrected the equation for variance. Returned values for variance and standard deviation were
previously wrong.

Added test cases for `vtkImageData.computeHistogram`.

<!--
👋 Hello, and thank you for starting this contribution!
📖 Make sure you've read our CONTRIBUTING.md guide before submitting your pull request.
❗️ Please follow the template below to help other contributors review your work.
-->

### PR and Code Checklist
<!--
NOTE: We will not merge if the following steps have not been completed!
-->
- [x] [semantic-release](https://github.com/semantic-release/semantic-release) commit messages
- [x] Run `npm run reformat` to have correctly formatted code

### Context
<!--
Explain why this change is needed. Please include relevant links supporting this change, such as:
- fix #ISSUE_NUMBER (from issue tracker)
- discourse post thread, or any other existing references
-->

### Changes
<!--
Please describe what is changing. Include:
- APIs added, deleted, deprecated, or changed
- Classes and methods added, deleted, deprecated, or changed
- A summary of usage if this is a new feature or change to an API. Adequate documentation and TS definitions should also be added/updated.
-->
- [x] Documentation and TypeScript definitions were updated to match those changes

### Results
<!--
Describe or illustrate the effects of your contribution. Please include:
- comparisons of the behavior before vs after
- screenshots of new or changed visualizations if applicable
-->

### Testing
<!--
Please describe how this can be tested by reviewers. Be specific about anything not tested and the reasons why. Tests should be added for new functionality and existing tests should complete without errors. See CONTRIBUTING.md
-->
- [x] This change adds or fixes unit tests
- [x] All tests complete without errors on the following environment:
  - **vtk.js**: <!-- ex: 14.0.0 (favor latest master) --> master
  - **OS**: <!-- ex: Windows 10, iOS 13.6 --> Windows 10 pro, version: 21H1 build: 19043.1288
  - **Browser**: <!-- ex: Chrome 89.0.4389.128 --> Chrome 95.0.4638.69

<!-- Remove the line below if it is not relevant -->
Attn: @agirault @floryst @jourdain

agirault commented:
Suggestion (not required): would ideally want those two changes (variance & count) to be different commits.

Are there tests for the histogram? 
* If not, could you add one?
* If yes, could you update it to show the variance was wrong, and is now right?

Ty

agirault commented on Sources/Common/DataModel/ImageData/index.js:
```diff
@@ -410,7 +410,7 @@ function vtkImageData(publicAPI, model) {
 
     const average = inum > 0 ? isum / inum : 0;
-    const variance = sumOfSquares - average * average;
+    const variance = Math.abs(sumOfSquares / inum - average * average);
```
@dzenanz do you mind reviewing that this is accurate? ty

dzenanz replied:
Taking absolute value is not needed, because sum of squares is guaranteed to be larger than square of average. But I believe the formula is now correct.

dzenanz replied:
I later realized numerical considerations were probably the reason for that.

floryst replied:
If we're checking the `inum > 0` case for the average calculation, best to do it here as well.

jadh4v replied:
Added check for `inum`, and a test case to cover that possibility.

jadh4v commented:
Done!
1. Created two separate commits for the bug fix and adding count to return.
2. Added test cases. There were no test cases for vtkImageData. Added a new file at Common/DataModel/ImageData/test/.

floryst commented:
LGTM

agirault reviewed (approved):
Ty 🚀
